### PR TITLE
Alston/share - ShareCard modal

### PR DESCRIPTION
**Pull request by alston-hsu**

### CODE REVIEW

File: packages/UI/src/components/organisms/cards/ShareCard/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export { default } from './ShareCard';
+
+export interface ShareCardProps {
+  title: 'idea' | 'position' | 'product';
+  platforms: string;
```
Comment (Enjoy2Live) [inline packages/UI/src/components/organisms/cards/ShareCard/index.ts:5]: @alston-hsu
We have specific `platforms` that we support in this regard, let's make this prop more limited by actually defining the strings one by one.

Edit: is this prop even used?

File: packages/UI/src/components/organisms/cards/ShareCard/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  link: string;
```
Comment (Enjoy2Live) [inline packages/UI/src/components/organisms/cards/ShareCard/index.ts:6]: @alston-hsu 
This prop isn't used right? if so let's remove it

File: packages/UI/src/components/organisms/cards/ShareCard/index.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  handleClose: () => void;
```
Comment (Enjoy2Live) [inline packages/UI/src/components/organisms/cards/ShareCard/index.ts:7]: @alston-hsu 
This prop isn't used right? if so let's remove it

File: packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+const shareIdea = (platform: string, ideaUrl: string) => {
+  const shareUrl = createShareUrl(platform, ideaUrl);
+
+  if (platform === 'mail') {
+    window.location.assign(shareUrl!);
+  } else {
+    window.open(shareUrl, '', 'width=500,height=500');
+  }
+};
```
Comment (Enjoy2Live) [inline packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts:35]: @alston-hsu
We need this to be a tag, doing this with JS is very bad in terms of accessibility

File: packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+const createShareUrl = (platform: string, ideaUrl: string) => {
+  const pageTitle = document.title;
+  const facebookShareUrl = 'https://www.facebook.com/sharer/sharer.php?u=';
+  const twitterShareUrl = 'https://twitter.com/intent/tweet?url=';
+  const mailShareUrl = `mailto:?&subject=${pageTitle}&cc=&bcc=&body=`;
```
[the remark this answers is not in the document]
Comment (alston-hsu) [reply packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts:5]: Added it

File: packages/UI/src/components/organisms/cards/ShareCard/index.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export { default } from './ShareCard';
+
+export interface ShareCardProps {
+  title: 'idea' | 'position' | 'product';
```
[the remark this answers is not in the document]
Comment (alston-hsu) [reply packages/UI/src/components/organisms/cards/ShareCard/index.ts]: Done

File: packages/UI/src/components/organisms/cards/ShareCard/index.ts (hunk 1 above)
Comment (alston-hsu) [reply packages/UI/src/components/organisms/cards/ShareCard/index.ts]: Done

File: packages/UI/src/components/organisms/cards/ShareCard/index.ts (hunk 2 above)
Comment (alston-hsu) [reply packages/UI/src/components/organisms/cards/ShareCard/index.ts]: Done

File: packages/UI/src/components/organisms/cards/ShareCard/index.ts (hunk 3 above)
Comment (alston-hsu) [reply packages/UI/src/components/organisms/cards/ShareCard/index.ts]: Done

File: packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (alston-hsu) [reply packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts]: Done

File: packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Enjoy2Live) [reply packages/UI/src/components/organisms/cards/ShareCard/ShareUrl.ts:5]: looks good, but it is missing the prop that consumers are going to use to fill in that body text.
with the current implementation it's always empty.

**Review by Enjoy2Live — CHANGES_REQUESTED** [review]

**Comment by Enjoy2Live** [comment]
closed as #1530 is being maintained better

**Pull request closed without merging**